pkg/daemon/rpm-ostree.go

```diff
@@ -148,6 +149,15 @@ func (r *RpmOstreeClient) GetBootedOSImageURL() (string, string, error) {
 		}
 	}
 
+	// we have container images now, make sure we can parse those too
+	if bootedDeployment.ContainerImageReference != "" {
+		// right now they start with "ostree-unverified-registry:", so scrape that off
```

Comment:
will this always be the case?

Reply:
I think at least for the time being we know there will be *a* prefix, but I don't know that it will always be explicitly the "ostree-unverified-registry" prefix. 

I can just scrape the prefix of "ostree-unverified-registry" and move on, but then if the prefix changes, we're broken. This here, however, will break if the prefix is ever completely absent, so that's not 100% future-proof either. 

We talked about this when we put this in the layering branch -- we were trying to avoid having to regex this, but I can?

Reply:
So...well, yes this is a bit messy.

It actually does open up the question of how the user specifies security-related settings for this image.  (Not just for the ostree-level signatures, but also e.g. TLS)

For the moment, I have to say that we hardcode injecting `ostree-unverified-registry:` as a prefix on the URL they provide.

We're planning to introduce a CRD, and that's probably the right place to have explicit settings for this.

That said, there is an argument that *the* interface to this `containers-policy.json` - and we shouldn't try to treat the OS image container specially at all.

But I dunno...it *is* different because we're going to boot it.

Anyways yeah my instinct right now is to inject `ostree-unverified-registry:` on the container image reference they provide.

Parsing the value by splitting on the first `:` is certainly something we can rely on right now.

But yes, this is all still technically experimental and change-able.  But we wouldn't change it without getting the MCO to work with the new way first!